includes/Classifai/Providers/Azure/TextToSpeech.php

```diff
@@ -0,0 +1,553 @@
+		$default_settings = $this->get_default_settings();
+		$voices_options   = $this->get_voices_select_options();
+
+		add_settings_field(
+			'url',
+			esc_html__( 'Endpoint URL', 'classifai' ),
+			[ $this, 'render_input' ],
+			$this->get_option_name(),
+			$this->get_option_name(),
+			[
+				'label_for'     => 'url',
+				'input_type'    => 'text',
+				'default_value' => $default_settings['url'],
+				'description'   => __( 'Supported protocol and hostname endpoints, e.g., <code>https://REGION.api.cognitive.microsoft.com</code> or <code>https://EXAMPLE.cognitiveservices.azure.com</code>. This can look different based on your setting choices in Azure.', 'classifai' ),
+			]
+		);
+
+		add_settings_field(
+			'api-key',
+			esc_html__( 'API Key', 'classifai' ),
+			[ $this, 'render_input' ],
+			$this->get_option_name(),
+			$this->get_option_name(),
+			[
+				'label_for'     => 'api_key',
+				'input_type'    => 'password',
+				'default_value' => $default_settings['api_key'],
+			]
+		);
+
+		add_settings_field(
+			'post-types',
+			esc_html__( 'Post Types', 'classifai' ),
+			[ $this, 'render_post_types_checkboxes' ],
+			$this->get_option_name(),
+			$this->get_option_name(),
+			[
+				'option_index' => 'post_types',
+			]
+		);
+
+		if ( ! empty( $voices_options ) ) {
+			add_settings_field(
+				'voice',
+				esc_html__( 'Voice', 'classifai' ),
+				[ $this, 'render_select' ],
+				$this->get_option_name(),
+				$this->get_option_name(),
+				[
+					'label_for'     => 'voice',
+					'options'       => $voices_options,
+					'default_value' => $default_settings['voice'],
+				]
+			);
+		}
+	}
+
+	/**
+	 * Sanitization callback for settings.
+	 *
+	 * @param array $settings The settings being saved.
+	 * @return array
+	 */
+	public function sanitize_settings( $settings ) {
+		$new_settings = $this->get_settings();
+		$new_settings = array();
+
+		if ( ! empty( $settings['url'] ) && ! empty( $settings['api_key'] ) ) {
+			$new_settings['url']     = trailingslashit( esc_url_raw( $settings['url'] ) );
+			$new_settings['api_key'] = sanitize_text_field( $settings['api_key'] );
+
+			$is_connected = $this->connect_to_service(
+				array(
+					'url'     => $new_settings['url'],
+					'api_key' => $new_settings['api_key'],
+				)
+			);
+
+			if ( is_wp_error( $is_connected ) ) {
+				$new_settings['authenticated'] = false;
+			} else {
+				$new_settings['authenticated'] = true;
+			}
+		} else {
+			$new_settings['url']           = '';
+			$new_settings['api_key']       = '';
+			$new_settings['authenticated'] = false;
+
+			add_settings_error(
+				$this->get_option_name(),
+				'classifai-azure-text-to-speech-auth-empty',
+				esc_html__( 'One or more credentials required to connect to the Azure Text to Speech service is empty.', 'classifai' ),
+				'error'
+			);
+		}
+
+		// Sanitize the post type checkboxes
+		$post_types = get_post_types( [ 'public' => true ], 'objects' );
```

Comment:
Couple things here. Looks like we're using `get_post_types_for_language_settings` to render out the checkbox options, seems like we should then utilize that here as well for consistency.

Also, I wasn't aware of this until Peter brought it up recently but instead of checking for `public => true`, he suggests using the `is_post_type_viewable` function (see https://github.com/10up/mention-notifier/pull/6#discussion_r1142811966). If we standardize to using the `get_post_types_for_language_settings` here, we should make that change there